lwgenerate/aws_controltower/aws_controltower.go

```diff
@@ -0,0 +1,455 @@
+	SqsQueueName string
+
+	// A map/dictionary of Tags to be assigned to created resources
+	Tags map[string]string
+
+	// Set this to true to use an existing IAM role from the log_archive AWS Account
+	UseExistingIamRole bool
+
+	// Amount of time to wait before the next resource is provisioned
+	WaitTime int
+
+	// The KMS key arn, if Control Tower was deployed with custom KMS key
+	KmsKeyArn string
+
+	// Mapping of AWS accounts to Lacework accounts within a Lacework organization
+	OrgAccountMappings OrgAccountMapping
+
+	// OrgAccountMapping json used for flag input
+	OrgAccountMappingsJson string
+
+	// Lacework Profile to use
+	LaceworkProfile string
+
+	// Lacework Organization
+	LaceworkOrganizationLevel bool
+
+	// The Lacework AWS Root Account ID
+	LaceworkAccountID string
+}
+
+type OrgAccountMapping struct {
+	DefaultLaceworkAccount string          `json:"default_lacework_account"`
+	Mapping                []OrgAccountMap `json:"mapping"`
+}
+
+type OrgAccountMap struct {
+	LaceworkAccount string   `json:"lacework_account"`
+	AwsAccounts     []string `json:"aws_accounts"`
+}
+
+func (args GenerateAwsControlTowerTfConfigurationArgs) GetSubAccounts() []AwsSubAccount {
+	return args.SubAccounts
+}
+
+func (args *GenerateAwsControlTowerTfConfigurationArgs) GetLaceworkProfile() string {
+	return args.LaceworkProfile
+}
+
+func (args *GenerateAwsControlTowerTfConfigurationArgs) Generate() (string, error) {
+	// Validate inputs
+	if err := args.validate(); err != nil {
+		return "", errors.Wrap(err, "invalid inputs")
+	}
+
+	// Create blocks
+	requiredProviders, err := createRequiredProviders()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate required providers")
+	}
+
+	awsProvider, err := createAwsProvider(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate aws provider")
+	}
+
+	laceworkProvider, err := createLaceworkProvider(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate lacework provider")
+	}
+
+	// controlTower
+	controlTowerModule, err := createCloudTrailControlTower(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate aws controlTower module")
+	}
+
+	// Render
+	hclBlocks := lwgenerate.CreateHclStringOutput(
+		lwgenerate.CombineHclBlocks(
+			requiredProviders,
+			awsProvider,
+			laceworkProvider,
+			controlTowerModule,
+		),
+	)
+	return hclBlocks, nil
+}
+
+type AwsSubAccount struct {
+	// The name of the AwsProfile to use (in AWS configuration)
+	AwsProfile string
+
+	// The AwsRegion this profile should use if any resources are created
+	AwsRegion string
+
+	// The Alias of the provider block
+	Alias string
+}
+
+func NewAwsSubAccount(profile string, region string, alias ...string) AwsSubAccount {
+	subaccount := AwsSubAccount{AwsProfile: profile, AwsRegion: region}
+	if len(alias) > 0 {
+		subaccount.Alias = alias[0]
+	}
+	return subaccount
+}
+
+func (args GenerateAwsControlTowerTfConfigurationArgs) validate() error {
```

Comment:
This is a good place to have the validation from `generate_aws_controltower.go` lines 158->188